contracts/wawaka/common/StringArray.cpp

```diff
@@ -64,44 +63,46 @@ StringArray::StringArray(const char* buffer)
 // XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
 StringArray::StringArray(const StringArray& value)
 {
-    size_ = value.size_;
-    value_ = new uint8_t[size_];
-    if (value_ == NULL)
-        return;
+    size_ = 0;
+    value_ = NULL;
 
-    memcpy(value_, value.value_, size_);
+    (void) assign(value.value_, value.size_);
 }
 
 // XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
 StringArray::~StringArray(void)
 {
-    if (value_ != NULL)
-        delete value_;
+    clear();
 }
 
 // XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
-bool StringArray::resize(const size_t size)
+bool StringArray::clear(void)
 {
     if (value_ != NULL)
-        delete value_;
+    {
+        free(value_);
+    }
 
-    size_ = size;
-    value_ = new uint8_t[size_];
-    if (value_ == NULL)
-        return false;
+    size_ = 0;
+    value_ = NULL;
 
     return true;
 }
 
 // XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
-bool StringArray::clear(void)
+bool StringArray::resize(const size_t size)
 {
-    if (value_ != NULL)
-        delete value_;
+    clear();
 
-    size_ = 0;
-    value_ = NULL;
+    if (size == 0)
```

Comment:
you really are going for the nits... where would you suggest we document that? and to what end?

Reply:
Where I was coming from with my comment was thinking about `resize`  having enlarge semantics for the array, rather than more generally reusing the same object to hold possibly entirely new data. So I guess that the size = 0 case being handled separately caught my attention. Like you said, this is really a nit, and I was really thinking that a short comment at the top of the function would be enough.

Reply:
resize is not intended to reuse any existing memory. it is really "clear and reallocate".

Reply:
comments added